packages/graph/hash_graph/lib/interval-ops/src/interval.rs

```diff
@@ -98,8 +98,10 @@ pub trait Interval<T>: Sized {
         let rhs_lower = other.lower_bound();
         let rhs_upper = other.upper_bound();
 
-        // Range A:    [-----] | [-----]
-        // Range B: [-----]    |    [-----]
+        // Example |      A     |     B
```

Comment:
I like a lot of the different format strategies in this commit, but I think we should try and be consistent with them. 

For instance, we differ between a heading of `Examples` and `Example`, we also differ between using numeric indices, or alphabetic ones. I'd probably encourage us to use numeric as you seem to be calling things "Range A" and "Range B" and such.